docs/command-line-syntax.md

```diff
@@ -0,0 +1,39 @@
+# How we document our command line syntax
+
+## Required arguments
+
+Use plain text
+
+_*example:*_
+`gh help`
+The argument help is required in this command
+
+## Placeholder values
+
+Use angled brackets to represent text that a user must supply
+
+_*example:*_
+`gh pr view <issueNumber>`
+Replace `<issueNumber>` with an issue number
+
+## Optional arguments
+
+Place optional arguments in square brackets
+
+_*example:*_
+`gh pr checkout [--web]`
+The argument `--web` is optional.
+
+## Mutually exclusive arguments
+
+Place mutually exclusive arguments inside braces, separate arguments with vertical bars.
+
+_*example:*_
+`gh pr {view | create}`
+
+## Repeatable arguements
+
+Ellipsis represent arguments that can appear multiple times
+
+_*example:*_
+`gh pr close <numbers … >`
```

Comment:
The ellipsis should be outside of angled brackets
```suggestion
`gh pr close <number>...`
```

Reply:
This is one where I had trouble. Google's example of using ellipsis looks like this.

> gcloud dns group [global-flag ...]

IBM has a [cli document](https://publib.boulder.ibm.com/tividd/td/tec/SC32-1232-00/en_US/HTML/ecormst15.htm) that shows ellipsis inside `[]` but outside `{}`. That kind of makes sense because `{}` is treated like a set, so the set is repeating.

So based on the docs and examples I see it looks like it goes inside angled brackets. Do you have examples of ellipses in existing cli apps we can look at?

Reply:
Inspecting the HTML source of google's document reveals ellipsis was outside of the placeholder:

```html
<code>
gcloud dns
<var>group</var>
[<var>global-flag</var> ...]
</code>
```

The `cat(1)` manual says:
```
cat [<OPTION>]... [<FILE>]...
```

`git-rev-parse(1)` says:
```
git rev-parse [<options>] <args>...​
```

AWS CLI says:
```
aws s3 <Command> [<Arg> ...]
```